### ISSUE TITLE
OpenMP codegen + bounds-checks = compiler error

### ISSUE THREAD

**Issue by manopapad** [StanfordLegion/legion#377]
When I run the following Regent source:

```
import "regent"

struct Radiation_columns {
  S : double;
}

local task source_term(points : region(ispace(int3d), Radiation_columns))
where
  reads writes (points.S)
do
  __demand(__openmp)
  for p in points do
    p.S = 0.0
  end
end

task work()
  var is = ispace(int3d, {64,64,64})
  var Radiation = region(is, Radiation_columns)
  var primColors = ispace(int3d, {2,2,2})
  var Radiation_primPart = partition(equal, Radiation, primColors)
  for color in primColors do
    source_term(Radiation_primPart[color])
  end
end

regentlib.start(work)
```

with:

```
regent.py test.rg -fopenmp 1 -fbounds-checks 1
```

I get:

```
src/terralib.lua:1123: Errors reported during typechecking.
/home/manolis/sw/legion/language/src/regent/codegen.t:880: definition of variable with symbol $runtime is not in scope in this context
        [cx.runtime], [cx.context],
          ^
/home/manolis/sw/legion/language/src/regent/codegen.t:880: definition of variable with symbol $context is not in scope in this context
        [cx.runtime], [cx.context],
                        ^
/home/manolis/sw/legion/language/src/regent/codegen.t:882: definition of variable with symbol $points is not in scope in this context
        [lr].impl, [region_index])
          ^
/home/manolis/sw/legion/language/src/regent/codegen.t:880: definition of variable with symbol $runtime is not in scope in this context
        [cx.runtime], [cx.context],
          ^
/home/manolis/sw/legion/language/src/regent/codegen.t:880: definition of variable with symbol $context is not in scope in this context
        [cx.runtime], [cx.context],
                        ^
/home/manolis/sw/legion/language/src/regent/codegen.t:882: definition of variable with symbol $points is not in scope in this context
        [lr].impl, [region_index])
          ^
/home/manolis/sw/legion/language/src/regent/codegen.t:880: definition of variable with symbol $runtime is not in scope in this context
        [cx.runtime], [cx.context],
          ^
/home/manolis/sw/legion/language/src/regent/codegen.t:880: definition of variable with symbol $context is not in scope in this context
        [cx.runtime], [cx.context],
                        ^
/home/manolis/sw/legion/language/src/regent/codegen.t:882: definition of variable with symbol $points is not in scope in this context
        [lr].impl, [region_index])
          ^

stack traceback:
	[C]: in function 'error'
	src/terralib.lua:388: in function 'finishandabortiferrors'
	src/terralib.lua:3314: in function 'typecheck'
	src/terralib.lua:1123: in function 'defineobjects'
	/home/manolis/sw/legion/language/src/regent/codegen.t:7143: in function 'fn'
	src/asdl.lua:18: in function 'block'
	/home/manolis/sw/legion/language/src/regent/codegen.t:9027: in function 'thunk'
	/home/manolis/sw/legion/language/src/regent/std_base.t:1123: in function 'complete'
	/home/manolis/sw/legion/language/src/regent/std.t:3528: in function 'compile_tasks_in_parallel'
	/home/manolis/sw/legion/language/src/regent/std.t:3635: in function 'start'
	test.rg:29: in main chunk
```

**Comment by magnatelee** [comment]
This should be fixed by 46cac201.

**Comment by lightsighter** [comment]
Does this mean we have bounds checks on OpenMP code now?

**Comment by magnatelee** [comment]
That's what I meant by "fixed".

**Comment by magnatelee** [comment]
BTW, this was really fixed by 8e97febc, which works around the thread safety issue in safe casting.

**Comment by manopapad** [comment]
Fixed
